versions/next/asyncapi.md

```diff
@@ -2208,6 +2231,68 @@ petstore_auth:
 - read:pets
 ```
 
+### <a name="correlationIdObject"></a>Correlation ID Object
+
+An object that specifies an identifier at design time that can used for message tracing and correlation.
+
+For specifying and computing the location of a Correlation ID a [runtime expression](#runtimeExpression) is used.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---|---
+description | `string` | A optional description of the identifier. [CommonMark syntax](http://spec.commonmark.org/) can be used for rich text representation.
+type | `integer` \| `string` \| `number` | An optional type specification.
+location | {expression} | **REQUIRED.** A runtime expression that specifies the location of the correlation ID.
+
+##### Examples
+
+```json
+{
+  "description": "Default Correlation ID",
+  "type": "string",
+  "location": "$message.header.correlationId"
+}
+```
+
+```yaml
+description: Default Correlation iD
+type: string
+location: $message.header.correlationId
+```
+
+### <a name="runtimeExpression"></a>Runtime Expression
+
+A runtime expression allows values to be defined based on information that will be available within the message.
+This mechanism is used by [Correlation ID Object](#correlationIdObject).
+
+The runtime expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
+
+```
+      expression = ( "$message" )
+      source = ( header-reference | payload-reference )  
+      header-reference = "header." token
+      payload-reference = "payload." name
+      fragment = a JSON Pointer [RFC 6901](https://tools.ietf.org/html/rfc6901) 
+      name = *( char )
+      char = as per RFC [7159](https://tools.ietf.org/html/rfc7159#section-7)
+      token = as per RFC [7230](https://tools.ietf.org/html/rfc7230#section-3.2.6)
+```
+
+The `name` identifier is case-sensitive, whereas `token` is not. 
+
+The table below provides examples of runtime expressions and examples of their use in a value:
+
+##### <a name="runtimeExpressionExamples"></a>Examples
+
+Source Location | Example expression  | Notes
+---|:---|:---|
+Message Header Property | `$message.header.correlationId` | Headers are analogous to message properties
```

Comment:
```suggestion
Message Header Property | `$message.header.correlationId` | Headers are analogous to message headers.
```

Reply:
What I was getting here is that some messaging providers call headers properties, so giving some guidance for the reader if they understand the meaning of properties not properties

Reply:
Yeah, but that should be explained in the message headers section, don't you think so? This note seems confusing to me because we never called them `message properties` (on purpose) and suddenly it appears here.

Reply:
Sure sounds good - will add something appropriate and then change the wording here

Reply:
Have changed the examples and hopefully made them more useful

Reply:
Indeed. Thanks!